Mathlib/Algebra/Order/Module/Defs.lean

```diff
@@ -203,12 +204,37 @@ class SMulPosReflectLT : Prop where
 namely `a₁ • b ≤ a₂ • b → a₁ ≤ a₂` if `0 < b`.
 
 You should usually not use this very granular typeclass directly, but rather a typeclass like
-`OrderedSMul`. -/
+`IsOrderedModule`. -/
 class SMulPosReflectLE : Prop where
   /-- Do not use this. Use `le_of_smul_lt_smul_right` instead. -/
   protected elim ⦃b : β⦄ (hb : 0 < b) ⦃a₁ a₂ : α⦄ (hb : a₁ • b ≤ a₂ • b) : a₁ ≤ a₂
 
 end Right
+
+section LeftRight
+variable [Zero α] [Zero β]
+
+/-- An ordered module is a module with a partial order such that scalar multiplication by a
+nonnegative scalar and of a nonnegative vector are both monotone. -/
+class IsOrderedModule where
+  /-- Do not use this. Use `smul_le_smul_of_nonneg_left` instead. -/
```

Comment:
Is there a particular reason why this doesn't just extend `PosSMulMono` and `SMulPosMono`, which is what one would expect when reading one of the docstrings further up in the diff?

Reply:
Yes, they have conflicting field names! That's an annoying oversight of mine back from when I defined these typeclasses. I would like to fix it, up to you on whether this happens before or after this PR, I'm easy.

Reply:
Oh I see. Feel free to do this in subsequent PR, I don't mind either.